Frends.AmazonS3.UploadObject/Frends.AmazonS3.UploadObject.Test/UnitTest1.cs

```diff
@@ -0,0 +1,916 @@
+                AwsSecretAccessKey = _secretAccessKey,
+                BucketName = _bucketName,
+                Region = Region.EuCentral1,
+            };
+            _options = new Options
+            {
+                UploadFromCurrentDirectoryOnly = false,
+                Overwrite = false,
+                PreserveFolderStructure = false,
+                ReturnListOfObjectKeys = false,
+                DeleteSource = false,
+                ThrowErrorIfNoMatch = false,
+            };
+
+            var result = AmazonS3.UploadObject(_connection, _input, _options, default);
+            Assert.IsNotNull(result.Result.Results);
+        }
+        #endregion AWSCredsACLLogDeliveryWrite
+
+        #endregion AWS Creds
+
+        #region HelperMethods
+
+        /// <summary>
+        /// Create test files and folders.
+        /// </summary>
+        [TestInitialize]
+        public void Initialize()
+        {
+            CreateTestFiles();
+        }
+
+        /// <summary>
+        /// Delete test files and folders.
+        /// </summary>
+        [TestCleanup]
+        public void CleanUp()
+        {
+            DeleteSourcePath();
+        }
+
+        /// <summary>
+        /// Create pre-signed URL
+        /// </summary>
+        /// <param name="key">File's name in S3</param>
+        /// <returns></returns>
+        private Uri CreatePresignedUrl(string key)
+        {
+            var region = RegionEndpoint.EUCentral1;
+            var client = new AmazonS3Client(_accessKey, _secretAccessKey, region);
+            GetPreSignedUrlRequest request = new()
+            {
+                BucketName = _bucketName,
+                Key = key,
+                Verb = HttpVerb.PUT,
+                Expires = DateTime.UtcNow.AddMinutes(15),
+
+            };
+            return new Uri(client.GetPreSignedURL(request));
+        }
+
+
+        /// <summary>
+        /// Test files and folders. Location Frends.AmazonS3.UploadObject.Test\bin\Debug\net6.0\AWS
+        /// </summary>
+        private void CreateTestFiles()
+        {
+            Directory.CreateDirectory($@"{_dir}\AWS");
+            Directory.CreateDirectory($@"{_dir}\AWS\Subfolder");
+
+            File.AppendAllText($@"{_dir}\AWS\test1.txt", "test1");
+            File.AppendAllText($@"{_dir}\AWS\Subfolder\subfile.txt", "From subfolder.");
+            File.AppendAllText($@"{_dir}\AWS\deletethis_presign.txt", "Resource file deleted. (Presign)");
+            File.AppendAllText($@"{_dir}\AWS\deletethis_awscreds.txt", "Resource file deleted. (AWS Creds)");
+            File.AppendAllText($@"{_dir}\AWS\overwrite_presign.txt", "Not overwriten. (Presign)");
+            File.AppendAllText($@"{_dir}\AWS\overwrite_awscreds.txt", "Not overwriten. (AWS creds)");
+        }
+
+        /// <summary>
+        /// Delete files from main folder to cause exeption in PreSignedNoFileInTopDirectoryTest().
+        /// </summary>
+        private void DeleteMainDirectoryFiles()
```

Comment:
Since this method is only used for one test, you should delete this method and move these steps to the test itself.